smart-contract-verifier/src/compiler/version.rs

```diff
@@ -10,50 +11,58 @@ pub enum ParseError {
     Parse(String),
     #[error("couldn't parse commit hash")]
     CommitHash(hex::FromHexError),
+    #[error("cannot parse semver: {0}")]
+    Semver(#[from] semver::Error),
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct ReleaseVersion {
     pub version: semver::Version,
-    pub commit: [u8; 4],
+    pub commit: String,
```

Comment:
I suggest to leave it as `[u8; 4]` but to add leading zeros in case if there is not enough bytes in the commit hash

Reply:
I did the exact same way you suggested, however there is some problems with that

+ what if in future we will have more that 4 bytes?
+ `v0.1.1+commit.12` and `v0.1.1+commit.1200` are the same, but what if they are not? 
+ we don't need this slice at all
So, I decided to just remove this slice, it just creates a lot of complications, but we don't need it

Reply:
Okay, let's leave it as a string